langkit/templates/python_api/module_py.mako

```diff
@@ -1856,3 +1857,67 @@ def _field_address(struct, field_name):
 #
 
 ${exts.include_extension(ctx.ext("python"))}
+
+#
+# App base classes
+#
+
+<% short_name = ctx.short_name.lower if ctx.short_name else 'mdl' %>
+
+class App(object):
+    """
+    Base class to regroup logic for an app. We use a class so that
+    specific languages implementations can add specific arguments and
+    processing by overriding:
+
+    - add_arguments to add arguments to the argparse.Parser instance
+
+    - create_unit_provider to return a custom unit provider to be used by the
+      AnalysisContext.
+    """
+
+    @property
+    def description(self):
+        """
+        Description for this app. Needs to be subclassed.
+        """
+        pass
```

Comment:
If it *needs* to be derived, I would recommend to raise a NotImplementedError exception in the body.